oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/HttpSessionOAuth2AuthorizationRequestRepository.java

```diff
@@ -113,14 +132,62 @@ private String getStateParameter(HttpServletRequest request) {
 	 * @return a non-null and mutable map of {@link OAuth2AuthorizationRequest#getState()}
 	 * to an {@link OAuth2AuthorizationRequest}.
 	 */
-	private Map<String, OAuth2AuthorizationRequest> getAuthorizationRequests(HttpServletRequest request) {
+	private Map<String, WrappedWithCreated<OAuth2AuthorizationRequest>> getAuthorizationRequests(
+			HttpServletRequest request) {
 		HttpSession session = request.getSession(false);
-		Map<String, OAuth2AuthorizationRequest> authorizationRequests = (session != null)
-				? (Map<String, OAuth2AuthorizationRequest>) session.getAttribute(this.sessionAttributeName) : null;
+		Map<String, WrappedWithCreated<OAuth2AuthorizationRequest>> authorizationRequests = (session != null)
+				? (Map<String, WrappedWithCreated<OAuth2AuthorizationRequest>>) session
+						.getAttribute(this.sessionAttributeName)
+				: null;
 		if (authorizationRequests == null) {
 			return new HashMap<>();
 		}
+		// remove expired entries
+		authorizationRequests.entrySet().removeIf((entry) -> entry.getValue().created
+				.isBefore(this.clock.instant().minus(this.oAuth2AuthorizationRequestExpiresIn)));
 		return authorizationRequests;
 	}
 
+	/**
+	 * Sets the {@link Clock} used in {@link Instant#now(Clock)} when setting the instant
+	 * created for {@link OAuth2AuthorizationRequest}.
+	 * @param clock the clock
+	 * @since 5.5
+	 */
+	public void setClock(Clock clock) {
+		Assert.notNull(clock, "clock cannot be null");
+		this.clock = clock;
+	}
+
+	/**
+	 * Sets the {@link Duration} for which {@link OAuth2AuthorizationRequest} should
+	 * expire.
+	 * @param oAuth2AuthorizationRequestExpiresIn the {@link Duration} a
+	 * {@link OAuth2AuthorizationRequest} is considered not expired. Must not be negative.
+	 * @since 5.5
+	 */
+	public void setOAuth2AuthorizationRequestExpiresIn(Duration oAuth2AuthorizationRequestExpiresIn) {
```

Comment:
Rename to `authorizationRequestTimeToLive()`

Reply:
:+1: